docs/workloads/azure/data/getting_started/etl_pipelines_deployment_azure.md

```diff
@@ -1,17 +1,123 @@
+
+Datastacks requires a config file for generating a new ingest workload. This config file should be a yaml file. A sample config file is included in the [de_templates](https://github.com/amido/stacks-azure-data/tree/main/de_templates) folder.
+
+Using the sample config file as a starting point, create a new config file and populate the values relevant to your new ingest pipeline, e.g.
+
+```yaml
+# `dataset_name` parameter is used to determine names of the following ADF resources:
+# - pipeline: Ingest_<dataset_name>
+# - dataset: ds_<dataset_name>
+# - linked service: ls_<dataset_name>
+dataset_name: AzureSql_MyNewExample
+pipeline_description: "Ingest from demo Azure SQL database using ingest config file."
+data_source_type: azure_sql
+
+key_vault_linked_service_name: ls_KeyVault
+data_source_password_key_vault_secret_name: sql-password
+data_source_connection_string_variable_name: sql_connection
+
+# Azure DevOps configurations
+
+ado_variable_groups_nonprod:
+  - amido-stacks-de-pipeline-nonprod
+  - amido-stacks-infra-credentials-nonprod
+  - stacks-credentials-nonprod-kv
+
+ado_variable_groups_prod:
+  - amido-stacks-de-pipeline-prod
+  - amido-stacks-infra-credentials-prod
+  - stacks-credentials-prod-kv
+
+# Datalake containers
+
+bronze_container: raw
+silver_container: staging
+gold_container: curated
+```
+
+## Generate project artifacts using Datastacks
+
+Use the Datastacks CLI to generate the artifacts for the new workload, using the prepared config file (replacing `path_to_config_file/my_config.yaml` with the appropriate path). Note, a workload with Data Quality steps requires a data platform with a Databricks workspace:
+
+```bash
+# Initiate Datastacks using poetry:
+poetry run datastacks
+
+# Generate resources for an ingest pipeline (without Data Quality steps)
+datastacks generate ingest --config="path_to_config_file/my_config.yaml"
+
+# Generate resources for an ingest pipeline (with added Data Quality steps)
+datastacks generate ingest --config="path_to_config_file/my_config.yaml" --data-quality
+```
+
+This should add new project artifacts for the workload under `de_workloads/ingest`, based on the ingest workload templates. Review the resources that have been generated.
+
+Once reviewed, commit the resources and push the branch to the remote project repository.
+
+## Deploy new workload in non-prod environment
+
+The generated workload will contain a YAML file containing a template Azure DevOps CI/CD pipeline for the workload, named `de-ingest-ado-pipeline.yaml`. This can be added as the definition for a new pipeline in Azure DevOps.
+1. Sign-in to your Azure DevOps organization and go to your project.
+2. Go to Pipelines, and then select New pipeline.
+3. Name the new pipeline to match the name of your new workload, e.g. `de-ingest-azuresql-mynewexample`
+4. For the pipeline definition, specify the YAML file in the project repository feature branch (e.g. `de-ingest-ado-pipeline.yaml`)
```

Comment:
I think at some point it is required to confirm access for the pipeline to the ADO variable groups, not sure when exactly though. @satenderrathee ?